packages/yarnpkg-fslib/sources/ZipFS.ts

```diff
@@ -864,18 +865,21 @@ export class ZipFS extends BasePortableFakeFS {
     return (attributes & S_IFMT) === S_IFLNK;
   }
 
-  private getFileSource(index: number) {
+  private getFileSource(index: number): Buffer
+  private getFileSource(index: number, asyncDecompress: boolean): Promise<Buffer>
+  private getFileSource(index: number, asyncDecompress: boolean = false): Promise<Buffer> | Buffer {
```

Comment:
I know it'll be a bit more annoying from a type perspective, but I'd prefer if it was an option map rather than a free argument - cf the boolean trap.